File: 2019/knowit/5/5.js
"use strict";

const fs = require("fs");
const R = require("ramda");

R.compose(
	console.log.bind(console),
	R.join(""),
	R.reverse,
	list => R.splitAt(list.length / 2, list),
	R.join(""),
	R.map(R.reverse),
	R.splitEvery(2),
	R.join(""),
	R.reverse,
	R.splitEvery(3),
	R.trim,
)(fs.readFileSync("./input.txt", "utf-8"));

File: 2019/knowit/5/5.hs
import System.IO (readFile)
import Data.List.Split (chunksOf)

main =
    print
        . concat
        . reverse
        . (\list -> chunksOf (length list `div` 2) list)
        . concat
        . map reverse
        . chunksOf 2
        . concat
        . reverse
        . chunksOf 3
        . filter (/= '\n')
        =<< readFile "input.txt"
